tests/application/cms/test_models.py

```diff
@@ -1,342 +1,361 @@
+import re
 from datetime import datetime, timedelta
 
 import pytest
 from bs4 import BeautifulSoup
 from flask import url_for
 
 from application.cms.exceptions import RejectionImpossible
-from application.cms.models import Dimension, DimensionClassification, MeasureVersion, UKCountry
-from tests.models import MeasureFactory, MeasureVersionFactory
-from tests.utils import create_measure_versions
+from application.cms.models import Dimension, UKCountry
+from tests.models import (
+    MeasureFactory,
+    MeasureVersionFactory,
+    MeasureVersionWithDimensionFactory,
+    ClassificationFactory,
+    EthnicityFactory,
+)
 
 
 class TestDimensionModel:
-    def test_create_valid_dimension(self, test_app_client, mock_rdu_user, stub_measure_page):
-        with test_app_client.session_transaction() as session:
-            session["user_id"] = mock_rdu_user.id
+    def test_create_valid_dimension(self, test_app_client, logged_in_rdu_user):
 
+        measure_version = MeasureVersionFactory(
+            version="1.0",
+            measure__slug="measure",
+            measure__subtopics__slug="subtopic",
+            measure__subtopics__topic__slug="topic",
+        )
```

Comment:
I think it might be nice not to pin these but just inject the generated values wherever we're using them later (in asserting the url?). What do you think?

Reply:
It makes the regex kind of ugly...
```
        assert re.match(rf"http://localhost:5000/cms/{measure_version.measure.subtopic.topic.slug}/"
                        rf"{measure_version.measure.subtopic.slug}/{measure_version.measure.slug}/"
                        rf"{measure_version.version}/[^/]+/edit",
                        resp.location)
```

But might be OK?